Fix evp SM cipherType check

gojimmypi opened the pull request:
# Description

wolfcrypt [evp.c](https://github.com/wolfSSL/wolfssl/blob/69450932211fbf0b5ff02f01d2858ff0e53e53c0/wolfcrypt/src/evp.c#L6042) appears to have an incorrect value for SM Ciphers during `ctx->cipherType` check. The gating `WOLFSSL_SM4_GCM` is used, whereas I believe they need to be the macros in this PR.

Edit: updates `wolfSSL_EVP_CIPHER_CTX_ctrl()` for `EVP_CTRL_AEAD_SET_TAG` and `EVP_CTRL_AEAD_GET_TAG` conditions when using SM Ciphers with `WOLFSSL_SM4_GCM` and/or `WOLFSSL_SM4_CCM` enabled.

Fixes zd# n/a

# Testing

How did you test?

Tested only on embedded Espressif device.

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

SparkiDev reviewed (changes requested):
CCM with SM4 as well.
Thanks.

gojimmypi commented:
I've updated the other section, not previously detected as there was a gating problem (also fixed).

There's also a spelling correction in an `ssl.c` comment included here.
